hpke/hpke.go

```diff
@@ -242,7 +242,11 @@ func (s *Sender) allSetup(rnd io.Reader) ([]byte, Sealer, error) {
 	case modeBase, modePSK:
 		enc, ss, err = k.EncapsulateDeterministically(s.pkR, seed)
 	case modeAuth, modeAuthPSK:
-		enc, ss, err = k.AuthEncapsulateDeterministically(s.pkR, s.skS, seed)
+		authKEM, ok := k.(kem.AuthScheme)
```

Comment:
What is this additional validation doing?

Reply:
It prevented from using a KEM that doesn't support authentication. (currently all kems do support).
updated now